.github/workflows/static_analysis.yaml

```diff
@@ -90,5 +90,40 @@ jobs:
 
       - name: Dead Code Analysis
         run: |
-          cd element-web 
+          cd element-web
           yarn run analyse:unused-exports
+
+  tsc-strict:
+    name: Typescript Strict Error Checker
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+      checks: write
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Get diff lines
+        id: diff
+        uses: Equip-Collaboration/diff-line-numbers@v1.0.0
+        with:
+          include: '["\\.tsx?$"]'
+
+      - name: Detecting files changed
+        id: files
+        uses: futuratrepadeira/changed-files@v3.2.1
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          pattern: '^.*\.tsx?$'
+
+      - uses: t3chguy/typescript-check-action@main
```

Comment:
from the other PR: this should probably be matrix-org

Reply:
This is a temporary thing until we have `--strict` - it is definitely not in a sane state for me to put it into `matrix-org` and its code style, with it being a fork, is very very different to all our other JS projects.

Reply:
Also will be trying to contribute my changes upstream in my own time, after which I'll switch over to the upstream repo for this action